pkg/services/ssosettings/ssosettingsimpl/service.go

```diff
@@ -99,6 +104,10 @@ func (s *Service) GetForProviderWithRedactedSecrets(ctx context.Context, provide
 		return nil, ssosettings.ErrNotConfigurable
 	}
 
+	if provider == social.SAMLProviderName && !s.licensing.FeatureEnabled(social.SAMLProviderName) {
+		return nil, ssosettings.ErrNotConfigurable
+	}
```

Comment:
I'd suggest to simplify this condition and make it more generic. Instead of checking the license every time you want to know if `SAML` is available as a provider you can build a list of available providers in the constructor function of the SSO Settings API Service. Then you can reuse the check from above `s.isProviderConfigurable(provider)` after you modify the `isProviderConfigurable()` func to match the provider in the list you built during the initialization phase of the service.

Reply:
Would a license expiration have an outdated state if we don't check for a valid license on demand?

Let's consider this scenario: a license could be valid while the constructor is being called (at startup), but it could expire the next day, right?

Reply:
That's a valid point, a license can expire while the SSO settings service is running.

This is related to how we check if SAML is enabled for the `Upsert()` and `Delete()` methods. Those are based on the `reloadables` list that contains pointers to all available providers. For SAML, we register the provider on startup and then in case the license is expired and SAML is not available anymore, the provider is still there and users can still modify or delete SAML via API. Or am I missing something here in case of update and delete for SAML?

Reply:
@dmihai Not at all. I've changed the SAML license validation upon initialization, as suggested.